src/protocols/aeternity/composables/aeNames.ts

```diff
@@ -0,0 +1,290 @@
+import { useAccounts } from '@/composables/accounts';
+import { Encoded } from '@aeternity/aepp-sdk';
+import { useTransactionList } from '@/composables/transactionList';
+import type {
+  ChainName,
+  IDefaultComposableOptions,
+  IName,
+  ITransaction,
+} from '@/types';
+import {
+  AUTO_EXTEND_NAME_BLOCKS_INTERVAL,
+  PROTOCOL_AETERNITY,
+  UPDATE_POINTER_ACTION,
+} from '@/constants';
+import { useAeSdk } from '@/composables/aeSdk';
+import { isInsufficientBalanceError } from '@/protocols/aeternity/helpers';
+import {
+  IAddressNamePair,
+  IAuction,
+  IAuctionBid,
+} from '@/types';
+import { useAeNetworkSettings } from '@/protocols/aeternity/composables/aeNetworkSettings';
+import { useModals } from '@/composables/modals';
+import { tg } from '@/store/plugins/languages';
+import {
+  createPollingBasedOnMountedComponents,
+  useStorageRef,
+} from '@/composables/composablesHelpers';
+
+const POLLING_INTERVAL = 10000;
+
+interface IUpdatePointerParams {
+  name: ChainName;
+  address?: any;
+  type?: keyof typeof UPDATE_POINTER_ACTION;
+}
+
+interface IAuctionEntryParams {
+  name: ChainName;
+  expiration: number;
+  bids: IAuctionBid[];
+}
+
+const ownedNames = useStorageRef<IName[]>([], 'owned-names');
+const pendingAutoExtendNames = ref<ChainName[]>([]);
+const areNamesFetching = ref(false);
+const defaults = ref<Record<string, ChainName>>({});
+const preferred = ref<Record<string, string>>({});
+const auctions = useStorageRef<Record<string, IAuction>>({}, 'names-auctions');
+
+const initPollingWatcher = createPollingBasedOnMountedComponents(POLLING_INTERVAL);
+
+export function useAeNames({ store }: IDefaultComposableOptions) {
+  const { openDefaultModal } = useModals();
+  const { aeActiveNetworkSettings } = useAeNetworkSettings();
+  const { fetchPendingTransactions } = useTransactionList({ store });
+
+  const {
+    nodeNetworkId,
+    getAeSdk,
+  } = useAeSdk({ store });
+
+  const {
+    isMiddlewareReady,
+    getMiddleware,
+    getMiddlewareRef,
+    fetchFromMiddlewareCamelCased,
+  } = useMiddleware();
+
+  const {
+    aeAccounts,
+    getLastActiveProtocolAccount,
+  } = useAccounts();
+
+  function getName(name: string): IName | undefined {
+    return ownedNames.value.find((ownedName) => ownedName.name === name);
+  }
+
+  function getDefaultName(address?: Encoded.AccountAddress) {
+    if (!address) {
+      return '';
+    }
+    return defaults.value ? defaults.value[`${address}-${nodeNetworkId.value}`] : '';
+  }
+
+  async function setPreferred(address: Encoded.AccountAddress) {
+    const response = await fetchJson(`${aeActiveNetworkSettings.value.backendUrl}/profile/${address}`).catch(() => {});
+
+    const { preferredChainName } = response;
+
+    const key = `${address}-${nodeNetworkId.value}`;
+    if (preferredChainName) {
+      preferred.value[key] = preferredChainName;
+    } else {
+      delete preferred.value[key];
+    }
+  }
+
+  function getPreferredName(address: Encoded.AccountAddress) {
```

Comment:
Instead of having 2 separate functions `getDefaultName` and `getPreferredName` I suggest to merge them into one `getName` which would check if provided address is a local address (`accounts` composable / `isLocalAccountAddress`) and returning owned name. In other case try to return external address name.